subsys/net/l2/wifi/wifi_native.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wifi_native, CONFIG_WIFI_NATIVE_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/wifi_native.h>
+
+sys_slist_t wifi_native_stacks = SYS_SLIST_STATIC_INIT(&wifi_native_stacks);
+
+void wifi_native_register_stack(struct wifi_native_stack *stack)
+{
+	sys_slist_append(&wifi_native_stacks, &stack->node);
+}
+
+struct wifi_native_mgd_iface *wifi_native_get_mgd_iface(struct net_if *iface)
+{
+	struct wifi_native_mgd_iface *mgd_iface;
+	struct wifi_native_stack *stack = NULL;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&wifi_native_stacks, stack, node) {
+		SYS_SLIST_FOR_EACH_CONTAINER(&stack->mgd_ifaces, mgd_iface, node) {
+			if (mgd_iface->iface == iface) {
+				return mgd_iface;
+			}
+		}
+	}
+
+	return NULL;
+}
+
+int wifi_native_register_mgd_iface(struct wifi_native_stack *stack, struct net_if *iface,
+				   const struct wifi_mgmt_ops *ops)
+{
+	struct wifi_native_mgd_iface *mgd_iface;
+
+	if (!stack || !iface || !ops) {
+		return -EINVAL;
+	}
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&stack->mgd_ifaces, mgd_iface, node) {
+		if (mgd_iface->iface == iface) {
+			return -EEXIST;
+		}
+	}
+
+	mgd_iface = k_malloc(sizeof(*mgd_iface));
+	if (!mgd_iface) {
+		return -ENOMEM;
+	}
```

Comment:
Do we really need to involve heap here, we usually aim to avoid that in the net stack. I'd prefer to see a preallocated array in `struct wifi_native_stack` instead, especially that such information on what interface will be managed by which supplicant could be easily determined build-time, we can't create network interfaces at runtime in Zephyr.

Reply:
I will convert linked lists for `stacks` and `mgd_ifaces` to arrays with configuration options.